### PR TITLE
executor: fix inl_hash_join hang when abnormal exit

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/54055
Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  Test issus   54055
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/join/index_lookup_join_test.go [hunk 1]
```diff
@@ -174,3 +174,26 @@ func TestIssue54688(t *testing.T) {
...
+	tk.MustExec("insert into s values(1), (128);")
+
+	tk.MustExec("set @@tidb_max_chunk_size=32;")
+	tk.MustExec("set @@tidb_index_join_batch_size=32;")
+	tk.MustExec("set @@tidb_index_lookup_join_concurrency=1;")
+
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/executor/join/testIssue54055_1", "2*return(false)->1*return(true)"))
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/executor/join/testIssue54055_2", "return(true)"))
```
Comment (XuHuaiyu) [inline pkg/executor/join/index_lookup_join_test.go:193]: Do we need to disable these 2 failpoint

**Review by XuHuaiyu — APPROVED** [review]

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**